passfunc.go

```diff
@@ -0,0 +1,38 @@
+package dns
+
+// PassFunc is used early in the server code to accept or reject a message. When rejecting the returned boolean is false.
+type PassFunc func(dh Header) (rcode int, ok bool)
```

Comment:
I can't say I really like the name. Though a better name doesn't immediately spring to mind.